fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+
+! set up ESMF time interval at center of iau window
+      call ESMF_TimeIntervalSet(IAU_offsetTI, h=iau_offset, rc=rc)
+!
+      allocate(originPetList(num_pes_fcst+wrttasks_per_group))
+      allocate(targetPetList(num_pes_fcst+wrttasks_per_group))
+!
+      k = num_pes_fcst
+      timerhs = mpi_wtime()
+      do i=1, write_groups
+
+! prepare petList for wrtComp(i)
+        lead_wrttask(i) = k
+        do j=1, wrttasks_per_group
+          petList(j) = k + j-1
+        enddo
+        k = k + wrttasks_per_group
+        last_wrttask(i) = k - 1
+!        if(mype==0)print *,'af wrtComp(i)=',i,'k=',k
+
+! prepare name of the wrtComp(i)
+        write(cwrtcomp,"(A,I2.2)") "wrtComp_", i
+! create wrtComp(i)
+        wrtComp(i) = ESMF_GridCompCreate(petList=petList, name=trim(cwrtcomp), rc=rc)
+!      print *,'af wrtComp(i)=',i,'name=',trim(cwrtcomp),'rc=',rc
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+! call into wrtComp(i) SetServices
+        call ESMF_GridCompSetServices(wrtComp(i), wrtSS, userRc=urc, rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+        if (ESMF_LogFoundError(rcToCheck=urc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
+
+! add configuration file
+        call ESMF_GridCompSet(gridcomp=wrtComp(i),config=CF,rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+! create wrtstate(i)
+        wrtstate(i) = ESMF_StateCreate(rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+! add the fcst FieldBundles to the wrtState(i) so write component can
+! use this info to create mirror objects
+        call ESMF_AttributeCopy(fcstState, wrtState(i), &
+                                attcopy=ESMF_ATTCOPY_REFERENCE, rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+        call ESMF_StateAdd(wrtState(i), fcstFB, rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+! call into wrtComp(i) Initialize
+        call ESMF_GridCompInitialize(wrtComp(i), importState=wrtstate(i), &
+                                     clock=clock_fv3, phase=1, userRc=urc, rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+        if (ESMF_LogFoundError(rcToCheck=urc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
+
+! remove fcst FieldBundles from the wrtState(i) because done with it
+        call ESMF_StateRemove(wrtState(i), fcstItemNameList, rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
```

Comment:
comment says that the fcst FieldBundle is going to be removed from wrtState(i) (added on line 699), but fcstItemNameList is actually removed, is that correct? I do not see where fcstItemNameList is added to wrtState.